### ISSUE TITLE
Modern PC Settings not searched?

### ISSUE THREAD

**Issue by vicjpapa** [Open-Shell/Open-Shell-Menu#57]
![settings1](https://user-images.githubusercontent.com/28926620/43658734-af7a3818-9727-11e8-8ce8-7dae8a4f90cf.png)

According to the previous forum the search function also search the modern PC settings.  Currently it's not showing up

![untitled](https://user-images.githubusercontent.com/28926620/43658761-c51b32d0-9727-11e8-911c-0673749f1764.png)

**Comment by ibuprophen1** [comment]
It's probably me but, I've never heard of the term "Modern PC Settings"... 

Are you referring to the specific Windows 10 PC Settings within the Windows Apps? 


~Ibuprophen

**Comment by vicjpapa** [comment]
I understand.  I also just stole the term from the the previous forum's Start Menu Search Guide[http://www.classicshell.net/forum/viewtopic.php?f=7&t=1588&sid=73121307cfe71d854cd013a12385367d](url)

![1](https://user-images.githubusercontent.com/28926620/43670766-eac484d8-975d-11e8-9e52-6d42f7bc971c.png)

Regarding what you're asking, I'm not really sure what to call it but seems like description is correct.  Here's another example of the difference in searching "Touch" in Classic vs. Default

![2](https://user-images.githubusercontent.com/28926620/43670783-1c298866-975e-11e8-9299-87ca718cb09f.png)

**Comment by coddec** [comment]
I have a feeling, it's due to changes made to windows 10, cause I'm still using version 4.3.1, which means it's not a bug introduced by new versions. Don't get me wrong, it's because I'm lazy and it still works... so I did not upgrade to latest dev builds...

**Comment by ibuprophen1** [comment]
I see now... 👍

This is similar/same as (i believe a combination of) the following 2 issues:

https://github.com/NeoClassic-UI/Menu/issues/4
https://github.com/NeoClassic-UI/Menu/issues/15

I believe that this is a known issue, especially, for Windows 10 users. 


~Ibuprophen

**Comment by XenHat** [comment]
The same framework is involved, but the issues are unrelated.

**Comment by Gittyperson** [comment]
I hope that when this resolved, the "old" settings will not be removed from the search results. Windows 10 still has duplicate setting controls and some are better handled in the old, Control Panel UI instead of the new "Settings" UI.

**Comment by ge0rdi** [comment]
As a workaround you can add following folder (and its subfolders) to Windows Indexing locations (`Control panel -> Indexing options`):

`%LOCALAPPDATA%\Packages\windows.immersivecontrolpanel_cw5n1h2txyewy\LocalState\Indexed\Settings`

`ClassicShell`/`OpenShell` uses Windows Indexing database to get `PC settings` list.

Though it seems standard Windows 10 start menu doesn't need this folder to be indexed. It shows modern Settings in search results by default.

So it should be possible to do this without explicitly adding the location to Indexing.
But currently I don't know how to achieve it ☹️

**Comment by vicjpapa** [comment]
My LocalState folder seems to be empty.

**Comment by ibuprophen1** [comment]
I've never heard of a LocalState folder as being a part of the Menu Directories... 


~Ibuprophen

**Comment by ge0rdi** [comment]
@vicjpapa 

What OS do you have?
Does official Windows start menu find those `Settings` items if you look for them?

**Comment by vicjpapa** [comment]
@ibuprophen1 sorry I don't understand what you meant

@ge0rdi Windows 10 pro.  Yes the official start menu can see it.  Most of the screenshots above are from my own desktop

**Comment by AveYo** [comment]
**I've made a quickfix for now**   

![Imgur](https://i.imgur.com/gns6Frk.png)

[Settings_to_StartMenu.bat](https://gist.github.com/AveYo/0238c8242df0421c685635a76ed678cb)  

Script generates Start Menu - Prgrams - Settings - ... entries from known ms-settings commands.  
Did not have the energy to research what .dll is used for localization, so it's hardcoded in English.
Results appear under Programs.  Better than nothing, right?

**Comment by DragoCubed** [comment]
I think #61 might be related

**Comment by anonimuos** [comment]
Up to Windows 10 1709, I can confirm that the Modern Settings search works. Does not work on 1809 and 1903. Have not tested 1803.

P.S. I was the original Classic Shell tester.

**Comment by anonimuos** [comment]
Well one easy fix is install StartIsBack++ (you don't have to use that Start menu or even buy a license). It installs some shell CLSIDs that are useful. After installing StartIsBack, open these from Run dialog or Explorer address bar:

shell:::{99E2B362-3E4E-4255-9B29-41A7F40777BA}
shell:::{99E2B362-3E4E-4255-9B29-41A7F40777BB}

Select all items and choose "Create shortcut". Let it create a shortcut to all of them on your Desktop. Now simply move the shortcuts to some folder inside the Start menu Programs folder and they will be searched by the original Classic Shell as well as Open Shell.

Here's Classic Shell searching them now:
https://i.imgur.com/4mDmcfM.png
![image](https://user-images.githubusercontent.com/46422834/65836335-48c71a80-e2a5-11e9-8fea-210a977416e6.png)

**Comment by alexolog** [comment]
I have classic shell on my desktop and open shell on the laptop.  Both are running Windows 10 1903 build 18362.  The classic shell shows PC settings, the open shell does not.

Maybe there are some windows settings that are different, but I don't know where to look.

**Comment by ge0rdi** [comment]
> I have classic shell on my desktop and open shell on the laptop. Both are running Windows 10 1903 build 18362. The classic shell shows PC settings, the open shell does not.

Most likely one of machines was upgraded from some older Win10 versions.
In such case you will have `.settingcontent-ms` files in `%LOCALAPPDATA%\Packages\windows.immersivecontrolpanel_cw5n1h2txyewy\LocalState\Indexed\Settings`.
These are indexed and searched by `Open-Shell` (or `Classic-Shell`).

Newer Win10 versions don't contain those files. So if you did clean installation, they aren't present and thus aren't searched.

You can copy those files to the other machine if you want quick workaround.

Alternatively you can use the `StartIsBack` hack described above.

**Comment by ltGuillaume** [comment]
Does not seem to work in W10 1709

**Comment by ge0rdi** [comment]
@ltGuillaume 
W10 1709 should use this new way of modern settings, will try to have a look.

BTW Is there anything that prevents you from upgrading to newer W10 version? 1709 will lose even enterprise support in October.

**Comment by ge0rdi** [comment]
@ltGuillaume 
I have installed build 4.4.150 on clean W10 1709 and it seems to work nicely:
![image](https://user-images.githubusercontent.com/2453094/91208141-0b177f80-e70a-11ea-8d17-d46048c9784d.png)

Press `Win+R` and type `shell:::{82E749ED-B971-4550-BAF7-06AA2BF7E836}`.
You should see folder with all modern settings. Do you see it?

**Comment by ltGuillaume** [comment]
Is Open-Shell supposed to create this clsid like StartIsBack does (see https://github.com/Open-Shell/Open-Shell-Menu/issues/57#issuecomment-536323665) on setup? That would make sense, since I manually updated by extracting the installer files. If so, can I trigger this manually?

**Comment by ge0rdi** [comment]
>  since I manually updated by extracting the installer files

This is definitely not recommended and not supported way of installing/upgrading Open-Shell.
For the future please just run installer and that's it.
You can even extract MSI file from EXE installer and install that (if you don't trust EXE for some reason).

To manually register new shell folder you need to do run `regsvr32 C:\Windows\System32\StartMenuHelper64.dll` from elevated command prompt.

**Comment by ltGuillaume** [comment]
I needed to do that again, yep! Works now, thanks!

**Comment by bonzibudd** [comment]
@ge0rdi 

Wow. This is a great day to be an Open-Shell user! This is probably the most needed addition in years, I am so happy this is fixed! I'm running Windows version 20H2 and it works flawlessly. I also see this as a great opportunity for people to switch from using Classic Shell (It seems like a lot of people still are) and even another reason to switch from the Win10 or 8 menu. I don't have the technical knowledge to know how you would make a shell location entangle with the program, but it seems like a similar solution to what StartIsBack++ has come up with. Big thanks to everyone helping to keep this program alive.

**Comment by ge0rdi** [comment]
> I don't have the technical knowledge to know how you would make a shell location entangle with the program, but it seems like a similar solution to what StartIsBack++ has come up with.

Yes. I was inspired by StartIsBack++.
Once I had virtual shell folder with modern settings it was quite easy to plug it into `Open-Shell`.

**Comment by anonimuos** [comment]
Not a big issue but can you look into why certain search terms don't produce a result in the Start menu (not even in Windows 10's own search) but if you directly search the Settings app, you get results?

For example I typed: 'microsoft account' (without quotes) into the Open Shell Start menu search box. No results (which is fine since the Windows 10 search also didn't show anything). But the Settings app shows the right result "Manage your account" page. Maybe it's something that can be improved further? Is the Settings app also searching further somewhere else?

Another example: I typed: ClearType. No results. But the Windows 10 Search/Cortana/Win+S shows 1 result. So does the Settings app. Where else is it searching?

I am on Windows 10 Enterprise, 1803 64-bit.

**Comment by ge0rdi** [comment]
@anonimuos 
> For example I typed: 'microsoft account' (without quotes) into the Open Shell Start menu search box.

Here is search result for `microsoft account` on my Win10 2004:
![image](https://user-images.githubusercontent.com/2453094/91700879-c874fd80-eb76-11ea-830e-4bb837987026.png)

So apparently you just need to update (quite old) Win10.

> Another example: I typed: ClearType. No results. But the Windows 10 Search/Cortana/Win+S shows 1 result.

Yup. The same for me.
Windows search seems to find `Adjust ClearType text` setting in Control panel.
I'll try to have a look why `Open-Shell` is not able to find it. Since it should search also Control panel settings.

**Comment by anonimuos** [comment]
Oh ok so first issue I can fix by upgrading to Windows 10 latest, which I will do maybe after 20H2 or whenever it's "worth it". Actually the reason I am on 1803 is I use multiple monitors and all Windows 10 versions after 1803 have a nasty bug with dual monitors in DWM or some other component (I suspect it's in DWM) where the wrong maximized window (the one behind the focused/active one) is incorrectly closed when you click the X button on the active one.

Despite posting the bug in Feedback Hub, they haven't fixed it for 5 releases which is why actually I am on 1803 Enterprise (which is still supported, not that I care about updates or support anyway).

Adjust ClearType text doesn't exist in my 1803 All Tasks Control Panel folder. Maybe that's why. But I am sure it was there in some early releases. I was comparing a list of shortcuts I made from StartIsBack's Settings shell namespace extension on older Windows 10 to the current folder added by Open Shell and I found some tasks missing/removed by Microsoft.

Maybe Settings app is also looking into some shortcuts which it has stored internally in some DLL, not the "All Control Panel tasks" CLSID folder.

**Comment by ge0rdi** [comment]
> Adjust ClearType text doesn't exist in my 1803 All Tasks Control Panel folder. Maybe that's why.

It is also not present in my `All Tasks Control Panel`. So apparently it is not part of `Control panel` as such.

> Maybe Settings app is also looking into some shortcuts which it has stored internally in some DLL, not the "All Control Panel tasks" CLSID folder.

I really don't know where exactly `Settings` app (and Windows search) looks for settings :(
But it definitely uses settings described in `AllSystemSettings_{253E530E-387D-4BC2-959D-E6F86122E5F2}.xml`.

And on my machine there is record for `cttune.exe` (seems this is the app that handles `Adjust ClearType text` setting) in that file.
So it should be possible to use it somehow in `Open-Shell`. Currently we are parsing just some of those settings (modern settings).

Will have a look at this.

**Comment by ge0rdi** [comment]
@anonimuos 
I've added support for more settings from `AllSystemSettings_{253E530E-387D-4BC2-959D-E6F86122E5F2}.xml`.
You can check in out in this [build](https://ci.appveyor.com/project/passionate-coder/open-shell-menu/builds/34951336/artifacts).

**Comment by anonimuos** [comment]
Great. Works on my Windows 10 2004 and 1803 (2 different PCs). Also I am trying all builds of Open Shell for now on Windows Windows 8.1 and Windows 7 for compatibility.

**Comment by ge0rdi** [comment]
> Also I am trying all builds of Open Shell for now on Windows Windows 8.1 and Windows 7 for compatibility.

Thank you 👍

**Comment by anonimuos** [comment]
Another adjacent issue that Microsoft has created - again, not a problem with Open Shell is: they have renamed "and" to & in Settings pages. Would it be possible for Open Shell to return results with '&' if the user types 'and' and vice versa? e.g. if I type "Apps and features" (due to muscle memory of Programs and Features), I get nothing.

**Comment by ge0rdi** [comment]
@anonimuos 
That's weird. Typing `apps and features` will show `Apps & Features` item for me.
Also looking for shorter  `apps features` works well.

**Comment by bonzibudd** [comment]
Hi @ge0rdi 

I found a small issue with the shell location. I noticed that because Open-Shell accesses that location for searches, it actually shows up in the "Frequent" section of File Explorer's jump list. 
![image](https://user-images.githubusercontent.com/61938331/94317885-41d4f580-ff55-11ea-83fd-859c4ba47cba.png)
When I click the location, Explorer freezes and restarts. Hopefully this can be fixed so that the folder doesn't show up in the jump list.
